VideoLocker/src/main/java/org/edx/mobile/course/CourseAPI.java

```diff
@@ -1,33 +1,50 @@
 package org.edx.mobile.course;
 
 import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
 import android.text.TextUtils;
 
 import com.google.inject.Inject;
 import com.google.inject.Singleton;
 
 import org.edx.mobile.http.RetroHttpException;
+import org.edx.mobile.model.api.ProfileModel;
+import org.edx.mobile.module.prefs.UserPrefs;
 
 import retrofit.RestAdapter;
-import retrofit.http.Query;
 
 @Singleton
 public class CourseAPI {
 
-    private CourseService courseService;
+    @NonNull
+    private final CourseService courseService;
+    @NonNull
+    private final UserPrefs userPrefs;
+
 
     @Inject
-    public CourseAPI(@NonNull RestAdapter restAdapter) {
+    public CourseAPI(@NonNull RestAdapter restAdapter, @NonNull UserPrefs userPrefs) {
         courseService = restAdapter.create(CourseService.class);
+        this.userPrefs = userPrefs;
     }
 
-    public @NonNull CourseList getCourseList(int page) throws RetroHttpException {
-        return courseService.getCourseList(page);
+    public
+    @NonNull
+    CourseList getCourseList(int page) throws RetroHttpException {
+        return courseService.getCourseList(getUsername(), null, true, page);
     }
 
-    public @NonNull CourseDetail getCourseDetail(@NonNull String courseId) throws RetroHttpException {
+    public
+    @NonNull
+    CourseDetail getCourseDetail(@NonNull String courseId) throws RetroHttpException {
         // Empty courseId will return a 200 for a list of course details, instead of a single course
         if (TextUtils.isEmpty(courseId)) throw new IllegalArgumentException();
-        return courseService.getCourseDetail(courseId);
+        return courseService.getCourseDetail(courseId, getUsername());
+    }
+
+    @Nullable
+    private String getUsername() {
+        final ProfileModel profile = userPrefs.getProfile();
+        return null == profile ? null : profile.username;
```

Comment:
When would `profile` ever be `null`?  Is that an expected case?  If so, the subsequent call to the Course API will fail, no?